src/frequenz/sdk/timeseries/_ringbuffer.py

```diff
@@ -0,0 +1,430 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Ringbuffer implementation with focus on time & memory efficiency."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from datetime import datetime, timedelta
+from typing import Any, Generic, List, Sequence, TypeVar
+
+import numpy as np
+from attr import dataclass
```

Comment:
I'd suggest to use `dataclass` from dataclasses, i.e.
`from dataclasses import dataclass`

Reply:
Ah. The auto-import did that for me, I didn't question it :sweat_smile: